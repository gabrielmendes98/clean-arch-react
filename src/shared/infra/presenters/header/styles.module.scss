@import 'shared/infra/styles/theme.scss';

.header {
  background-color: $paper-color;
  color: $primary-color;
}
