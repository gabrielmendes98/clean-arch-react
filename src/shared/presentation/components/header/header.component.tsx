import { Link } from 'react-router-dom';
import { PAGES } from 'shared/domain/constants/pages';
import styles from './header.module.scss';

export const Header = () => (
  <header className={styles.header}>
    <nav>
      <ul>
        <li>
          <Link to={PAGES.HOME}>Home</Link>
        </li>
        <li>
          <Link to={PAGES.REGISTER_EMPLOYEES}>Cadastrar Funcionario</Link>
        </li>
        <li>
          <Link to={PAGES.LIST_EMPLOYEES}>Lista de Funcionarios</Link>
        </li>
      </ul>
    </nav>
  </header>
);
