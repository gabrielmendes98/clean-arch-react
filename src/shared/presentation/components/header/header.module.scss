@import 'shared/presentation/styles/theme.scss';

.header {
  background-color: $paper-color;
  color: $primary-color;
}
