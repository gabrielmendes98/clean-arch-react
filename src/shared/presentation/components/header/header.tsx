import style from './header.module.scss';

export const Header = () => (
  <header className={style.header}>
    <nav>
      <ul>
        <li>Home</li>
        <li>Cadastrar Funcionario</li>
        <li>Lista de Funcionarios</li>
      </ul>
    </nav>
  </header>
);
