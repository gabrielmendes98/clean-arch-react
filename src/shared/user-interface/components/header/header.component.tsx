import { memo } from 'react';
import { Link } from 'react-router-dom';
import { pages } from 'shared/domain/config/pages';
import styles from './header.module.scss';

export const HeaderComponent = () => (
  <header className={styles.header}>
    <nav>
      <ul>
        <li>
          <Link to={pages.home}>Home</Link>
        </li>
        <li>
          <Link to={pages.login}>Login</Link>
        </li>
        <li>
          <Link to={pages.signUp}>Registrar-se</Link>
        </li>
        <li>
          <Link to={pages.registerEmployees}>Cadastrar Funcionario</Link>
        </li>
        <li>
          <Link to={pages.listEmployees}>Lista de Funcionarios</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export const Header = memo(HeaderComponent);
