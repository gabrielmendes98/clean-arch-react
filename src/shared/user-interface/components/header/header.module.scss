@import 'shared/user-interface/styles/theme.scss';

.header {
  background-color: $paper-color;
  color: $primary-color;
}
