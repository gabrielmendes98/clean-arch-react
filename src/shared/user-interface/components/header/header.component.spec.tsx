import { MemoryRouter } from 'react-router-dom';
import { render } from 'shared/testing/test-utils';
import { Header } from './header.component';

describe('Header', () => {
  it('should navigate to pages', async () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
