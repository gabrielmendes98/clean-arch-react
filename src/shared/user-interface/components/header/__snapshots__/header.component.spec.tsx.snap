// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should navigate to pages 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/login"
          >
            Login
          </a>
        </li>
        <li>
          <a
            href="/sign-up"
          >
            Registrar-se
          </a>
        </li>
        <li>
          <a
            href="/register-employees"
          >
            Cadastrar Funcionario
          </a>
        </li>
        <li>
          <a
            href="/employees"
          >
            Lista de Funcionarios
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
